mlflow/utils/databricks_utils.py

```diff
@@ -439,34 +475,65 @@ def get_databricks_host_creds(server_uri=None):
         talk to the Databricks server.
     """
+    # Since we do not record OAuth expiration time in OAuth file, perform periodic refresh
+    # of OAuth cache here. As currently configured (02/24) OAuth token in model serving environment
+    # guaranteed to have at least 30 min remaining on TTL at any point in time but refresh at higher
+    # rate here to be safe and in case those values change in the future.
+    OAUTH_CACHE_REFRESH_DURATION = 10 * 60
+    OAUTH_CACHE_ENV_VAR = "DATABRICKS_DEPENDENCY_OAUTH_CACHE"
+    OAUTH_CACHE_EXPIRATION_ENV_VAR = "DATABRICKS_DEPENDENCY_OAUTH_CACHE_EXIRY_TS"
+
     profile, path = get_db_info_from_uri(server_uri)
     config = ProfileConfigProvider(profile).get_config() if profile else get_config()
-    # if a path is specified, that implies a Databricks tracking URI of the form:
-    # databricks://profile-name/path-specifier
-    if (not config or not config.host) and path:
-        dbutils = _get_dbutils()
-        if dbutils:
-            # Prefix differentiates users and is provided as path information in the URI
-            key_prefix = path
-            host = dbutils.secrets.get(scope=profile, key=key_prefix + "-host")
-            token = dbutils.secrets.get(scope=profile, key=key_prefix + "-token")
-            if host and token:
-                config = DatabricksConfig.from_token(host=host, token=token, insecure=False)
-    if not config or not config.host:
-        _fail_malformed_databricks_auth(profile)
-
     insecure = hasattr(config, "insecure") and config.insecure
 
-    if config.username is not None and config.password is not None:
-        return MlflowHostCreds(
-            config.host,
-            username=config.username,
-            password=config.password,
-            ignore_tls_verification=insecure,
-        )
-    elif config.token:
-        return MlflowHostCreds(config.host, token=config.token, ignore_tls_verification=insecure)
-    _fail_malformed_databricks_auth(profile)
+    # if in model serving environment databricks attempt to fetch model dependency OAuth token
+    if is_in_databricks_model_serving_environment():
+        # check if dependency is cached in env var before reading from file
+        oauth_token = ""
+        if (
+            OAUTH_CACHE_ENV_VAR in os.environ
+            and OAUTH_CACHE_EXPIRATION_ENV_VAR in os.environ
+            and float(os.environ[OAUTH_CACHE_EXPIRATION_ENV_VAR]) > time.time()
+        ):
+            oauth_token = os.environ[OAUTH_CACHE_ENV_VAR]
+        else:
+            try:
+                oauth_token = _get_model_dependency_oauth_token()
+                os.environ[OAUTH_CACHE_ENV_VAR] = oauth_token
+                os.environ[OAUTH_CACHE_EXPIRATION_ENV_VAR] = str(
+                    time.time() + OAUTH_CACHE_REFRESH_DURATION
+                )
+            except Exception as e:
+                _fail_model_serving_creds_env(e)
```

Comment:
```suggestion
            try:
                oauth_token = _get_model_dependency_oauth_token()
            except Exception as e:
                _fail_model_serving_creds_env(e)
                
            os.environ[OAUTH_CACHE_ENV_VAR] = oauth_token
            os.environ[OAUTH_CACHE_EXPIRATION_ENV_VAR] = str(
                time.time() + OAUTH_CACHE_REFRESH_DURATION
            )
```

I'd pull this out from try-catch because they are unlikely to fail.